src/auto/style_context.rs

```diff
@@ -398,6 +416,16 @@ impl<O: IsA<StyleContext>> StyleContextExt for O {
     //    unsafe { TODO: call gtk_sys:gtk_style_context_get_style() }
     //}
 
+    fn get_style_property(&self, property_name: &str, value: &mut glib::Value) {
+        unsafe {
+            gtk_sys::gtk_style_context_get_style_property(
+                self.as_ref().to_glib_none().0,
+                property_name.to_glib_none().0,
+                value.to_glib_none_mut().0,
```

Comment:
May be this function better be manual as 
`fn get_style_property(&self, property_name: &str) -> glib::Value`

Reply:
not sure that it can return `Option`, same as `gtk_style_context_get_property` at minimum by docs

Reply:
What would it return if the property name is invalid though?

Reply:
Docs don't say that in both cases :(

Reply:
Code says `g_critical()`, so we would cause memory unsafety here. We need to check the existence of the property beforehand somehow... like in `Object::get_property()`

Reply:
This is a style property though and not a regular GObject property. Looking at the StyleContext API I don't see any way to check beforehand.

Reply:
We could have manual code that initializes the `GValue` with `mem::zeroed()` and afterwards checks if the type of it is still 0.

Reply:
Looking into it, this function is deprecated (though not documented as such). Maybe we could ignore this one, and fix `get_property`?

Reply:
Sure, works for me. Since when is the other function deprecated?

Reply:
`get_property` isn't deprecated, and I'm not sure when this one was deprecated. Someone mentioned it in IRC as deprecated, and sure enough it [isn't in the GTK4 docs](https://developer.gnome.org/gtk4/stable/GtkStyleContext.html#gtk-style-context-get-property)

Reply:
Maybe they meant that it's deprecated as in "does not exist in GTK4 anymore".

I think for the "deprecated" one the way forward is to panic if it fails (the `g_critical()` from C is basically a panic), or otherwise make it return a `Result`.

The other one, if we can check beforehand then let's do that or otherwise like above.